Automattic/wp-calypso
# Add presentation attribute for iframe

* Closes https://github.com/Automattic/wp-calypso/issues/75601

## Proposed Changes

This PR adds the `role` attribute to `iframe` with `sharing-buttons-preview-class` to tell the screen readers how to handle that `iframe`. 

## Testing Instructions

* Navigate to `https://wordpress.com/marketing/sharing-buttons/{site}
* Make sure you have **Official Buttons** option selected in the **Buttons style** like so:

<img width="914" alt="Screenshot 2023-05-22 at 3 56 45 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/496b5d31-7468-483b-8de3-208c9cee6f45">

This will ensure that the `iframe` in question appears.
* Inspect the `iframe` with Chrome dev tools and check that the `role: presentation` is present:

<img width="1251" alt="Screenshot 2023-05-22 at 3 59 19 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/b3f2b2be-3bcd-41f1-bd95-ca6fae03c62c">

* If you have axeDevTools installed, run the scan of the page. Check that there is no messages related to that specific `iframe`:

<img width="1252" alt="Screenshot 2023-05-22 at 4 01 10 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/3ec45c07-1258-47d2-9202-cfd71ce5e6fc">

This was the warning that was happening before this PR:


<img width="1239" alt="Screenshot 2023-05-22 at 4 02 11 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/cf60c7df-9788-4c1a-99fd-774e04205c22">


## Pre-merge Checklist
